src/systemtest/java/com/ericsson/ei/systemtest/utils/JenkinsManager.java

```diff
@@ -0,0 +1,215 @@
+
+public class JenkinsManager {
+
+    private static final String JENKINS_JOB_TEMPLATE_FILE_PATH = String.join(File.separator, "src", "systemtest",
+            "resources", "jenkinsJobTemplate.xml");
+    private static final String DEFAULT_JOB_TOKEN = "test";
+    private static final String DEFAULT_SCRIPT = "echo &quot;Test&quot;";
+    private static final String DEFAULT_JOB_NAME = "test_job";
+
+    private String host;
+    private int port;
+    private String encoding;
+    private String crumb;
+
+    /**
+     * Constructor, takes jenkins host, port, username and password.
+     * @param host
+     * @param port
+     * @param username
+     * @param password   **  Jenkins password or API-token
+     * @throws UnsupportedEncodingException
+     * @throws URISyntaxException
+     * @throws JSONException
+     */
+    public JenkinsManager(String host, int port, String username, String password)
+            throws UnsupportedEncodingException, URISyntaxException, JSONException {
+        String authString = String.join(":", username, password);
+
+        this.host = host;
+        this.port = port;
+        this.encoding = DatatypeConverter.printBase64Binary(authString.getBytes("utf-8"));
+        this.crumb = getCrumb();
+    }
+
+    /**
+     * Takes a token and script and returns XML data for jenkins with job_token and
+     * script added.
+     *
+     * @param job_token
+     * @param script
+     * @return
+     * @throws IOException
+     */
+    public String getXmlJobData(String job_token, String script) throws IOException {
+        String jobData = getStringFromFile(JENKINS_JOB_TEMPLATE_FILE_PATH);
+
+        if (job_token == null) {
+            job_token = DEFAULT_JOB_TOKEN;
+        }
+        if (script == null) {
+            script = DEFAULT_SCRIPT;
+        }
+
+        jobData = injectScriptAndTokenIntoXmlString(jobData, job_token, script);
+        return jobData;
+    }
+
+    /**
+     * Creates a jenkins job with a given name using the XML data as input for
+     * job configuration
+     *
+     * @param jobName
+     * @param jobXmlData
+     * @return
+     * @throws URISyntaxException
+     * @throws ClientProtocolException
+     * @throws IOException
+     */
+    public boolean createJob(String jobName, String jobXmlData) throws URISyntaxException {
+        HttpRequest httpRequest = new HttpRequest(HttpMethod.POST);
+        boolean success = false;
+
+        if (jobName == null) {
+            jobName = DEFAULT_JOB_NAME;
+        }
+
+        httpRequest.setHost(host).setPort(port).addHeader("Authorization", "Basic " + encoding)
+                .addHeader("Content-type", MediaType.APPLICATION_XML_VALUE).addHeader("Jenkins-Crumb", crumb)
+                .addParam("name", jobName).setBody(jobXmlData).setEndpoint("/createItem");
+
+        ResponseEntity<String> response = httpRequest.performRequest();
+        success = response.getStatusCode() == HttpStatus.OK;
```

Comment:
True or false doesn't say that much,,, isn't better to also have a try catch block here and log the real fault as well. Then we can log which http error code and message received from Jenkins.

Reply:
There is nothing to catch here however we could log the statuscode etc if !success.

Reply:
done